kolena/metrics/_geometry.py

```diff
@@ -98,7 +97,13 @@ def as_shapely_polygon(obj: Union[BoundingBox, Polygon]) -> ShapelyPolygon:
 
 
 def _inf_with_iou(inf: Inf, iou_val: float) -> Inf:
-    return dataclasses.replace(inf, iou=iou_val)  # type: ignore[call-arg]
+    input_args = {
+        key: value
+        for key, value in inf._to_dict().items()
+        if key not in ["width", "height", "area", "aspect_ratio", "iou", "data_type"]
+    }
+    out: Inf = inf.__class__(**input_args, iou=iou_val)  # type: ignore[call-arg,assignment]
```

Comment:
Any issue with just using `inf.__class__(**inf._to_dict(), iou=iou_val)`?

Reply:
We will still have to filter out `iou` since it may already be defined on the input inference and cause an error. `data_type` would also become exposed as a property, when it ordinarily wouldn't be.

No real issue with passing the other properties - just seemed a bit strange to pass them explicitly when they're going to be derived from the bbox coords anyways.

Reply:
My concern is hard coding fields here, and this might lead to unintentional drift. How about this

```
def _inf_with_iou(inf: Inf, iou_val: float) -> Inf:
    exclude = ["iou"] + [f.name for f in dataclasses.fields(inf) if not f.init]
    obj = {k: v for k, v in vars(inf).items() if k not in exclude}
    out = inf.__class__(**obj, iou=iou_val)
    return out
```

side note: not impacting the use case here, but `_to_dict` is mainly used for serialization, so technically, if a member is also a dataobject, the returned value would also be a serialized version instead of the original object.